ext/fileinfo/fileinfo.stub.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+class finfo
+{
+    function __construct(int $options = FILEINFO_NONE, ?string $arg = null);
+
+    /** @return string|false */
+    function file(?string $file_name = null, int $options = FILEINFO_NONE, $context = null);
+
+    /** @return string|false */
+    function buffer(?string $string = null, int $options = FILEINFO_NONE, $context = null);
+
+    /** @return bool */
+    function set_flags(int $options);
+}
+
+/** @return resource|false */
+function finfo_open(int $options = FILEINFO_NONE, ?string $arg = null) {}
+
+function finfo_close($finfo): bool {}
+
+function finfo_set_flags($finfo , int $options): bool {}
+
+/** @return string|false */
+function finfo_file($finfo, ?string $file_name = null, int $options = FILEINFO_NONE, $context = null) {}
+
+/** @return string|false */
+function finfo_buffer($finfo, ?string $string = null, int $options = FILEINFO_NONE, $context = null) {}
+
+/** @return string|false */
+function mime_content_type(string $filename) {}
```

Comment:
This function accepts an arbitrary zval.

Reply:
Because of this right?
https://github.com/php/php-src/blob/d04c85fcb5977e56615e021327782720d997c3c6/ext/fileinfo/fileinfo.c#L447

Reply:
Exactly. The actually supported types (i.e. string and resource) can be added as doc-block @param annotation.

Reply:
Are we adding `@param` annotations? I couldn't see any when I looked originally so presumed it was only `@return` we were doing

Reply:
Well, some PRs have added them; guess it's okay at least.

Reply:
I've added them to all functions here